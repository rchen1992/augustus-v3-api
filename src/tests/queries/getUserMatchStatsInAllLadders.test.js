const mockUserRepo = require('@tests/mockRepos/user');
const { testQuery } = require('@tests/utils');

testQuery({
    id: "Get a user's match stats in all of their ladders",
    query: `
        {
            user(id: 1) {
                id
                userLadders {
                    id
                    matchStats {
                        matchCount
                        winCount
                        lossCount
                        tieCount
                    }
                    ladder {
                        id
                        ladderName
                    }
                }
            }
        }
    `,
    variables: {},
    context: () => {
        return {
            repos: {
                user: mockUserRepo,
            },
            services: {
                match: {
                    getLadderUserMatchStats() {
                        return {
                            matchCount: 7,
                            winCount: 4,
                            lossCount: 2,
                            tieCount: 1,
                        };
                    },
                },
            },
        };
    },
    expected: undefined,
});
