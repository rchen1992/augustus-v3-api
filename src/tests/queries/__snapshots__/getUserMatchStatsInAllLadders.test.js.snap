// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Get a user's match stats in all of their ladders 1`] = `
Object {
  "data": Object {
    "user": Object {
      "id": "1",
      "userLadders": Array [
        Object {
          "id": "1:1",
          "ladder": Object {
            "id": "1",
            "ladderName": "Tagpad",
          },
          "matchStats": Object {
            "lossCount": 2,
            "matchCount": 7,
            "tieCount": 1,
            "winCount": 4,
          },
        },
        Object {
          "id": "2:1",
          "ladder": Object {
            "id": "2",
            "ladderName": "Kamba",
          },
          "matchStats": Object {
            "lossCount": 2,
            "matchCount": 7,
            "tieCount": 1,
            "winCount": 4,
          },
        },
        Object {
          "id": "3:1",
          "ladder": Object {
            "id": "3",
            "ladderName": "Skyvu",
          },
          "matchStats": Object {
            "lossCount": 2,
            "matchCount": 7,
            "tieCount": 1,
            "winCount": 4,
          },
        },
        Object {
          "id": "4:1",
          "ladder": Object {
            "id": "4",
            "ladderName": "Thoughtworks",
          },
          "matchStats": Object {
            "lossCount": 2,
            "matchCount": 7,
            "tieCount": 1,
            "winCount": 4,
          },
        },
        Object {
          "id": "5:1",
          "ladder": Object {
            "id": "5",
            "ladderName": "Yakijo",
          },
          "matchStats": Object {
            "lossCount": 2,
            "matchCount": 7,
            "tieCount": 1,
            "winCount": 4,
          },
        },
      ],
    },
  },
}
`;
